sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Emitter.cs

```diff
@@ -115,25 +121,17 @@ private void AddFunctionMetadataInfo(IndentedTextWriter indentedTextWriter, IRea
                 }
             }
 
-            private void AddBindingInfo(IndentedTextWriter indentedTextWriter, string functionVarName, string functionBindingsListVarName, IList<IDictionary<string, string>> bindings)
+            private void AddBindingInfo(IndentedTextWriter indentedTextWriter, string functionVarName, string functionBindingsListVarName, IList<IDictionary<string, object>> bindings)
             {
                 var bindingCount = 0;
 
                 foreach (var binding in bindings)
                 {
                     var bindingVarName = functionVarName + "binding" + bindingCount.ToString();
-                    indentedTextWriter.WriteLine($"var {bindingVarName} = new {{");
-                    indentedTextWriter.Indent++;
-                    
-                    foreach (var key in binding.Keys)
-                    {
-                        indentedTextWriter.WriteLine($"{key} = {binding[key]},");
-                    }
 
-                    indentedTextWriter.Indent--;
-                    indentedTextWriter.WriteLine("};");
-                    indentedTextWriter.WriteLine($"var {bindingVarName}JSON = JsonSerializer.Serialize({bindingVarName});");
-                    indentedTextWriter.WriteLine($"{functionBindingsListVarName}.Add({bindingVarName}JSON);");
+                    var jsonBinding = JsonSerializer.Serialize(binding, _jsonOptions);
+
+                    indentedTextWriter.WriteLine($"{functionBindingsListVarName}.Add($\"{{{jsonBinding.Replace("\"", "\\\"")}}}\");");
```

Comment:
Would be helpful to see tests with varied data to see how this works. Is there any input that would escape this string and break compilation?